To: vim_dev@googlegroups.com
Subject: Patch 7.3.045
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.045
Problem:    Compiler warning for uninitialized variable.
Solution:   Initialize the variable always.
Files:	    src/getchar.c


*** ../vim-7.3.044/src/getchar.c	2010-10-27 12:17:54.000000000 +0200
--- src/getchar.c	2010-10-27 17:28:45.000000000 +0200
***************
*** 3290,3298 ****
       */
      if (haskey)
  	keys = replace_termcodes(keys, &keys_buf, TRUE, TRUE, special);
      if (hasarg)
      {
- 	orig_rhs = rhs;
  	if (STRICMP(rhs, "<nop>") == 0)	    /* "<Nop>" means nothing */
  	    rhs = (char_u *)"";
  	else
--- 3290,3298 ----
       */
      if (haskey)
  	keys = replace_termcodes(keys, &keys_buf, TRUE, TRUE, special);
+     orig_rhs = rhs;
      if (hasarg)
      {
  	if (STRICMP(rhs, "<nop>") == 0)	    /* "<Nop>" means nothing */
  	    rhs = (char_u *)"";
  	else
*** ../vim-7.3.044/src/version.c	2010-10-27 17:11:11.000000000 +0200
--- src/version.c	2010-10-27 17:30:11.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     45,
  /**/

-- 
MARTHA'S WAY: Don't throw out all that leftover wine. Freeze into ice cubes
              for future use in casseroles and sauces.
MY WAY:       What leftover wine?

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
